tests/unit/amlb/resources/test_framework_definition.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+from amlb.resources import Resources
+from amlb.utils import Namespace as NS
+
+
+def test_framework_definition_lookup_is_case_insensitive():
+    res = NS(
+        _frameworks=NS(
+            lower=NS(name="lower"),
+            UPPER=NS(name="UPPER"),
+            Camel=NS(name="CaMel")
+        )
+    )
+    # binding `framework_definition` method to our resource mock: use pytest-mock instead?
```

Comment:
we will most probably need `mock` or `pytest-mock` very soon, but keeping this for a future PR

Reply:
Namespace appears very useful to create mock objects here :)

Reply:
Any particular reason for not using @pytest.mark.parametrize so each condition is checked (and can fail) separately? Similar to other tests.

Reply:
I'll change, but I find parametrized tests much harder to read and less flexible (big ojects like resources need to be rebuilt each time) with few benefits in most cases.